lib/io/export.ex

```diff
@@ -30,28 +31,47 @@ defmodule Plsm.IO.Export do
   """
   @spec prepare(Plsm.Database.Table, String.t) :: String.t
   def prepare(table, project_name) do
-      output = module_declaration(project_name,table.header.name) <> model_inclusion() <> primary_key_declaration(table.columns) <> schema_declaration(table.header.name)
-      trimmed_columns = remove_foreign_keys(table.columns)
-      column_output = trimmed_columns |> Enum.reduce("",fn(x,a) -> a <> type_output({x.name, x.type}) end)
-      output = output <> column_output
-      belongs_to_output = Enum.filter(table.columns, fn(column) ->
-        column.foreign_table != nil and column.foreign_table != nil
-      end)
-      |> Enum.reduce("",fn(column, a) ->
-        a <> belongs_to_output(project_name, column)
-      end)
-      output = output <> belongs_to_output <> "\n"
-
-      output = output <> two_space(end_declaration())
-      output = output <> changeset(table.columns)
-      output <> end_declaration()
-  end
-
-  @spec primary_key_declaration([Plsm.Database.Column]) :: String.t
+    output = module_declaration(project_name,table.header.name) <> model_inclusion() <> primary_key_declaration(table.columns) <> schema_declaration(table.header.name)
+    trimmed_columns = case number_of_primary_keys(table.columns) do
+      x when x in 0..1 -> remove_foreign_keys(table.columns) |> remove_primary_keys |> remove_id_column
+      _ -> remove_foreign_keys(table.columns)
+    end
+    column_output = trimmed_columns |> Enum.reduce("",fn(x,a) -> a <> type_output({x.name, x.type}) <> primary_key_field?(x) end)
+    output = output <> column_output
+    belongs_to_output = Enum.filter(table.columns, fn(column) ->
+      column.foreign_table != nil and column.foreign_table != nil
+    end)
+    |> Enum.reduce("",fn(column, a) ->
+      a <> belongs_to_output(project_name, column)
+    end)
+    output = output <> belongs_to_output <> "\n"
+
+    output = output <> two_space(end_declaration())
+    output = output <> changeset(table.columns)
+    output <> end_declaration()
+  end
+
+  defp primary_key_field?(field) do
+    case field.primary_key do
+      true -> ", primary_key: true\n"
+      _ -> "\n"
+    end
+  end
+
+
+  def number_of_primary_keys(columns) do
+    Enum.filter(columns, &(&1.primary_key))
+      |> Enum.count
+  end
+
   defp primary_key_declaration(columns) do
-    Enum.reduce(columns, "", fn(x,acc) -> acc <> create_primary_key(x)  end)
+    keys = Enum.filter(columns, &(&1.primary_key))
+    case Enum.count(keys) do
+      1 -> create_primary_key(List.first(keys))
```

Comment:
There can be multiple primary keys on a table and it appears that this will only catch the first one.